robowflex_library/src/robot.cpp

```diff
@@ -838,8 +838,14 @@ bool Robot::setFromIK(const IKQuery &query)
     return setFromIK(query, *scratch_);
 }
 
-bool Robot::setFromIK(const IKQuery &query, robot_state::RobotState &state) const
+bool Robot::setFromIK(const IKQuery &q, robot_state::RobotState &state) const
 {
+    // copy query for unconstness
+    IKQuery query(q);
```

Comment:
Don't copy the entire query, just copy the tip frame and only the first one (if the array is longer than length 0).